### ISSUE TITLE
Add possibilty to create table WITH defined compression

### ISSUE THREAD

**Issue by fiedukow** [prestodb/presto#7826]
`CREATE TABLE` should allow not only to choose file format but also compression codec. It seems like it can be added to `WITH` clause. Currently (as far as I understand) it can be only be changed by changing global server property `hive.compression-codec`.

Is there any technical reason why it is not there yet?

**Comment by dain** [comment]
It is something we talked about.  The problem is Hive does't really have a place to support this and instead relies on the (IMO, lame) precedence set by Hadoop MR where each job can set a different compression.  When we have talked about this, we discussed adding a new Hive table property to and store this information in the Hive metastore, but this is further complicated by the fact that some formats now have custom compression configuration properties (like ORC).  This means the implementation would need to map the property set by the user to the correct property for the format, and if the format didn't have one use some Presto property.  Then for INSERT and SHOW CREATE TABLE, you have to reverse the whole process.... so no technical reason, other than it is a bunch of work.

**Comment by neymanm** [comment]
Hi,
There is a technical reason for that. We extract billions of records from our tables by creating tables with text format in order to send them to our customers. They, in tern, have to load this data into their databases, and cannot accept proprietary formats like ORC. In fact, they request text files with BZ2 compression. The quickest way to extract data would be using Presto. However, because you can specify neither compression type nor file sizes in Presto we have to use Hive, which is much slower.
It might be beneficial to have default compression that would generate splitable files rather than Gzip, so it's easier to use extracted data elsewhere. 
If it is a problem adding it to a table property, would it be easier to add compression type as a server configuration parameter or session level parameter (set session...), similar to Hive?

**Comment by findepi** [comment]
@neymanm, as long as adding compression type as a server configuration parameter works for you, i.e. you're OK with same value for every new table, you can use `hive.compression-codec`  property of Hive connector (https://prestodb.io/docs/current/connector/hive.html).

**Comment by neymanm** [comment]
Thank you. It may be very helpful. However, any global change will need extensive analysis and testing for possible impact. 
I am wondering if it's feasible to add it as a session parameter as well?

**Comment by neymanm** [comment]
Hi Piotr,
As I learned from Amazon support, you cannot specify BZ2 compression in Presto at all. Even on the server level using hive.compression-codec. Here is the response from Amazon support:
Testing with various combinations of BZip2 resulted in the below error(as you have already mentioned):
Could not coerce value 'BZip2' to com.facebook.presto.hive.HiveCompressionCodec (property 'hive.compression-codec') in order to call [public com.facebook.presto.hive.HiveClientConfig com.facebook.presto.hive.HiveClientConfig.setHiveCompressionCodec(com.facebook.presto.hive.HiveCompressionCodec)]

Also, I have observed that it accepts only one of the below mentioned values
NONE
SNAPPY
GZIP

Then checking the source code of HiveCompressionCodec in Presto, I can see that it agrees with my observation.
https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/HiveCompressionCodec.java
https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/HdfsConfigurationUpdater.java

public enum HiveCompressionCodec
{
    NONE(null, CompressionKind.NONE, CompressionCodecName.UNCOMPRESSED),
    SNAPPY(SnappyCodec.class, CompressionKind.SNAPPY, CompressionCodecName.SNAPPY),
    GZIP(GzipCodec.class, CompressionKind.ZLIB, CompressionCodecName.GZIP);

    private final Optional<Class<? extends CompressionCodec>> codec;
    private final CompressionKind orcCompressionKind;
private final CompressionCodecName parquetCompressionCodec;

Having said this, as of now, you will only be able to set SNAPPY or GZIP compression by "hive.compression-codec" in /etc/presto/conf/catalog/hive.properties.
This looks to me a good feature request to Presto to support BZip2 compression type..
